### PR TITLE
[intl] speed up localization generation and regenerate symbols

### PR DESCRIPTION
**Pull request by jonahwilliams**
Cupertino localizations shows up in both the startup CPU and memory profile for customer money. This isn't actually because loading cupertino localizations is that slow, but since cupertino localizations loads first it triggers the parsing of the date symbols table from a const map. Skip this work by instead declaring the date symbols using the intl.DateSymbols class directly.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
waiting to see if this triggers a test somewhere...

**Comment by jiahaog** [comment]
Nice, I was tracking this in b/225289302 too but I'll assign it to you then :)

File: dev/tools/localization/bin/gen_date_localizations.dart [hunk 1]
```diff
@@ -123,28 +133,67 @@ Future<void> main(List<String> rawArgs) async {
   if (writeToFile) {
     final File dateLocalizationsFile = File(path.join('packages', 'flutter_localizations', 'lib', 'src', 'l10n', 'generated_date_localizations.dart'));
     dateLocalizationsFile.writeAsStringSync(buffer.toString());
-    Process.runSync(path.join('bin', 'cache', 'dart-sdk', 'bin', 'dartfmt'), <String>[
```
Comment (jonahwilliams) [inline dev/tools/localization/bin/gen_date_localizations.dart:126]: This no longer exists

**Comment by jonahwilliams** [comment]
Would like some guidance on this. There appear to be no tests, but realy this is all an implementation detail anyway. Love to get your thoughts

File: dev/tools/localization/bin/gen_date_localizations.dart [hunk 2]
```diff
@@ -87,16 +87,26 @@ Future<void> main(List<String> rawArgs) async {
...
 
+import 'package:intl/date_symbols.dart' as intl;
+
 '''
 );
   buffer.writeln('''
 /// The subset of date symbols supported by the intl package which are also
 /// supported by flutter_localizations.''');
-  buffer.writeln('const Map<String, dynamic> dateSymbols = <String, dynamic> {');
+  buffer.writeln('final Map<String, intl.DateSymbols> dateSymbols = <String, intl.DateSymbols> {');
```
Comment (HansMuller) [inline dev/tools/localization/bin/gen_date_localizations.dart:97]: final implies lazy initialization like late final?

File: dev/tools/localization/bin/gen_date_localizations.dart [hunk 3]
```diff
@@ -123,28 +133,67 @@ Future<void> main(List<String> rawArgs) async {
   if (writeToFile) {
     final File dateLocalizationsFile = File(path.join('packages', 'flutter_localizations', 'lib', 'src', 'l10n', 'generated_date_localizations.dart'));
     dateLocalizationsFile.writeAsStringSync(buffer.toString());
-    Process.runSync(path.join('bin', 'cache', 'dart-sdk', 'bin', 'dartfmt'), <String>[
-      '-w',
+    final String extension = Platform.isWindows ? '.exe' : '';
```
Comment (HansMuller) [inline dev/tools/localization/bin/gen_date_localizations.dart:136]: Seems odd that this has to be done "by hand"

**Review by HansMuller — APPROVED** [review]
This seems OK to me. Even if the performance difference was a wash, this seems like a better way to deal with the Date symbols..

File: dev/tools/localization/bin/gen_date_localizations.dart (hunk 1 above)
Comment (goderbauer) [reply dev/tools/localization/bin/gen_date_localizations.dart:126]: We may be missing a test here that should have failed when this was removed?

**Review by goderbauer — APPROVED** [review]
Presumably, we have tests in place that cover using the dateSymbols? If those are still passing:

LGTM

File: dev/tools/localization/bin/gen_date_localizations.dart (hunk 3 above)
Comment (jonahwilliams) [reply dev/tools/localization/bin/gen_date_localizations.dart:136]: We wrote package:process to handle this back in the day

**Comment by jonahwilliams** [comment]
Yeah we don't appear to be running this on CI. Maybe we should, then we could check against a hash or something?

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Comment by jonahwilliams** [comment]
This led to some size reductions on small apps, which is surprising IMO.

https://mail.google.com/mail/u/0/#starred/FMfcgzGpFgvPrpRbQkPRlkDVCzpZBcXs

**Comment by jonahwilliams** [comment]
https://golem.corp.goog/PerformanceChanges?repository=flutter&revision=130124

**Pull request merged**